---
title: "R Notebook"
output: html_notebook
---

Now that I'm back to thinking family patterns relate to crashes, I'd like to identify some taxa that are higher in 3 or 5 day treatments.

Lasso didn't work, probably because we are under sampled. I think I'll try t-tests

First run SecondPassCore.Rmd
```{r}
library(infer) # tidy t-test
library(qvalue)
```


T identify how frequently things show up in the dataset (even with abundance zero)

```{r}
TTestInquire <- metabeast %>%
  filter(taxlevel %in% c("Family", "Genus")) %>%
  select(-comm) %>%
  unnest(df) %>%
  group_by(taxlevel, DayN, Taxid) %>%
  summarise(n = n(), .groups = "keep") %>%
  arrange(-n)
TTestInquire
```

Combined data processing
```{r}

TTestPreamble <- metabeast %>%
  #filter(taxlevel %in% c("Family", "Genus")) %>%
  select(-comm) %>%
  unnest(df) %>%
  filter(DayN <= 12) %>%
  left_join(dayCat, by = "DayN") %>%
  mutate(ratio = sqrt(ratio)) %>% # Testing normalization
  identity()

TTHits <- TTestPreamble %>%
  group_by(taxlevel, DayCat, Taxid) %>%
  summarise(appears_in = sum(ratio > 0)) %>%
  arrange(-appears_in) %>%
  ungroup()

TTestPreamble_01 <- TTestPreamble %>%
  left_join(TTHits, by = join_by(taxlevel, Taxid, DayCat)) %>%
  filter(appears_in >=5) # Was 6

TTestInsanity <- TTestPreamble_01 %>%
  group_by(taxlevel, DayCat, Taxid) %>%
  nest() %>%
  mutate(mod = map(data, ~t_test(., ratio~Info1, order = c("Crash", "Good")))) %>% 
  select(-data) %>%
  unnest(mod)
```
Why do some species have only some observations? What the heck?

```{r}
TTestInsanity %>%
  filter(!is.na(p_value)) %>%
  filter(DayCat == "3-5") %>%
  arrange(p_value) %>%
  head()
```

```{r}
TTestSig0 <- TTestInsanity %>%
  ungroup() %>%
  filter(!is.na(p_value)) %>%
  group_by(DayCat) %>%
  mutate(FDR_ByDay = p.adjust(p_value, method = "BH")) %>%
  #mutate(q_ByDay = qvalue(p_value)$qvalues) %>% # Not enough data
  ungroup() %>%
  mutate(FDR_Ungrouped = p.adjust(p_value, method = "BH")) %>%
  mutate(q_Ungrouped = qvalue(p_value)$qvalues) %>%
  identity()

str(TTestSig0)

TTestSig <- TTestSig0 %>%
  filter(p_value < 0.05) %>%
  filter(!is.na(Taxid))

dim(TTestSig)
TTestSig %>% arrange(DayCat)
```

What is going on with my false discovery rates and why are they 0.381 regardless of P-value?

https://www.ncbi.nlm.nih.gov/nucleotide/LC075346.1?report=genbank&log$=nuclalign&blast_rank=1&RID=R70Z0NHG013

## I want to show the bugs that have 3-5 day significance

```{r}
library(forcats)
Dinoflaggelata_Df <- hatch %>%
  group_taxa(Phylum) %>%
  filter(Phylum == "Dinoflagellata") %>%
  rename(ID = Phylum) %>%
  #select(ID, Sample0, Info1, `Brood#`, ratio) %>%
  identity()

Dinophyceae_Df <- hatch %>%
  group_taxa(Class) %>%
  filter(Class == "Dinophyceae") %>%
  rename(ID = Class) %>%
  #select(ID, Sample0, Info1, `Brood#`, ratio) %>%
  identity()

ASV_3_5_Df <- hatch %>%
  filter(nASV %in% c(53, 205)) %>%
  rename(ID = ASV) %>%
  #select(ID, Sample0, Info1, `Brood#`, ratio) %>%
  identity()

All_3_5_Df <- bind_rows(
  Dinoflaggelata_Df,
  Dinophyceae_Df,
  ASV_3_5_Df
) %>%
  mutate(ID = factor(ID, levels = c("Dinoflagellata", "Dinophyceae", "ASV_53", "ASV_205"))) %>%
  mutate(Info1 = forcats::fct_relevel(Info1, c("Good", "Crash"))) %>%
  identity()

All_3_5_Df
```

```{r}
All_3_5_Df %>%
  filter(ID != "Dinoflagellata") %>%
ggplot(aes(x = as.numeric(DayN), y = log10(ratio + 1/1000), shape = Info1, color = Info1, group = `Brood#`)) +
  geom_point(size = 2) + scale_color_manual(values = c("darkblue", "red")) +
  geom_path() +
  scale_x_continuous(breaks = c(seq(from = 3, to = 21, by = 2))) +
  facet_wrap(~ID, nrow = 2) +
  labs(x = "Larval Age (Days)",y = expression(log[10](frac("Microbe 16S", "Host 18S"))), color = "Batch Outcome", shape = "Batch Outcome") +

  theme_bw() +
  theme(legend.position = c(0.8, 0.25),
        legend.box.background = element_rect(linewidth = 1))
```

```{r fig.height = 2.25, fig.width = 7}
All_3_5_Df %>%
  filter(ID != "Dinoflagellata") %>%
ggplot(aes(x = as.numeric(DayN), y = log10(ratio + 1/1000), shape = Info1, color = Info1, group = `Brood#`)) +
  geom_point(size = 2) + scale_color_manual(values = c("darkblue", "red")) +
  geom_path() +
  scale_x_continuous(breaks = c(seq(from = 3, to = 21, by = 2))) +
  facet_wrap(~ID, nrow = 1) +
  labs(x = "Larval Age (Days)",y = expression(log[10](frac("Microbe 16S", "Host 18S"))), color = "Batch Outcome", shape = "Batch Outcome") +

  theme_bw() +
  theme(legend.position = c(0.875, 0.7),
        legend.box.background = element_rect(linewidth = 1))
```

## Nice looking table of statisticlly significant hits

```{r}
SigFT <- TTestSig %>%
  arrange(DayCat, taxlevel) %>%
  mutate(taxlevel = recode(taxlevel, "Tag_ASV" = "ASV")) %>%
  mutate(Taxid = recode(Taxid, "Mitochondria;53" = "ASV;53", "Pirellulaceae;205" = "ASV;205")) %>% # Because I think blast is more informative for these and the classifier was sort of wrong.
  select(Timepoint = DayCat, `Taxonomic Level`=  taxlevel, `Taxonomic Classification` = Taxid, `T` = statistic, p = p_value) %>% # FDR is always 0.19, and q = 0.03, so lets say so in the text and 
  mutate(across(`T`:p, ~formatC(.x, digits = 2, format = "g", flag = "#"))) %>%
  flextable()
SigFT
```

```{r}
flextable::save_as_docx(SigFT, path = here("Tables", "SignificantTaxa.docx"))
```

